Packs/Base/Scripts/DBotTrainClustering/DBotTrainClustering.py

```diff
@@ -0,0 +1,823 @@
+        """
+        self.clustering = clustering  # type: Type[Clustering]
+        self.threshold = threshold  # type: float
+        self.generic_cluster_name = generic_cluster_name
+        self.stats = {}  # type: ignore
+        self.statistics()
+        self.compute_dist()
+        self.date_training = datetime.now().strftime("%m/%d/%Y %H:%M:%S")
+        self.summary = None  # type: ignore
+        self.json = None  # type: ignore
+
+    def statistics(self):
+        """
+        Compute statistics of the clusters
+        """
+        # plot_silhouette = self.com_silhouette()
+        self.stats['General'] = {}
+        self.stats['General']['Nb sample'] = self.clustering.raw_data.shape[0]  # type: ignore
+        self.stats['General']['Nb cluster'] = self.clustering.number_clusters
+        self.stats['General']['min_samples'] = self.clustering.model.min_samples  # type: ignore
+        self.stats['General']['min_cluster_size'] = self.clustering.model.min_cluster_size  # type: ignore
+        for number_cluster in range(0, self.clustering.number_clusters):  # type: ignore
+            self.stats[number_cluster] = {}
+            self.stats[number_cluster]['number_samples'] = sum(
+                self.clustering.model.labels_ == number_cluster)  # type: ignore
+            ind = np.where(self.clustering.model.labels_ == number_cluster)[0]  # type: ignore
+            selected_data = [x for x in self.clustering.raw_data.iloc[ind][FAMILY_COLUMN_NAME]]  # type: ignore
+            counter = collections.Counter(selected_data)
+            total = sum(dict(counter).values(), 0.0)
+            dist = {k: v * 100 / total for k, v in counter.items()}
+            dist = dict((k, v) for k, v in dist.items() if v >= 1)
+            self.stats[number_cluster]['distribution sample'] = dist
+
+    def compute_dist(self):
+        """
+        Compute distribution of sample per cluster (depending of the naming and threshold)
+        """
+        dist_total = {}  # type: Dict
+        if not self.generic_cluster_name:
+            for cluster_number in range(0, self.clustering.number_clusters):  # type: ignore
+                chosen = {k: v for k, v in self.stats[cluster_number]['distribution sample'].items() if
+                          v >= self.threshold * 100}
+                if not chosen:
+                    continue
+                total = sum(dict(chosen).values(), 0.0)
+                dist = {k: v * 100 / total for k, v in chosen.items()}
+                dist_total[cluster_number] = {}
+                dist_total[cluster_number]['number_samples'] = sum(
+                    self.clustering.raw_data[  # type: ignore
+                        self.clustering.model.labels_ == cluster_number].label.isin(list(chosen.keys())))  # type: ignore
+                dist_total[cluster_number]['distribution'] = dist
+                dist_total[cluster_number]['clusterName'] = ' , '.join([x for x in chosen.keys()])
```

Comment:
I think it's better to take into account when generating names based on incidents fields values:
1. Trim long names
2. Perhaps differentiate between clusters with same name with a number

Reply:
1. ok
2. It's fine to have cluster with the same name. No problem. It's just an indication of the content of the cluster

Reply:
we have also colors - so not sure we need numbers
Triming long names it's a good idea - @yguezpa please trim names to 16 charecters (add "..." if trimmed)